pages/auth/signin.tsx

```diff
@@ -1,27 +1,52 @@
 "use client";
 
-import type { GetServerSidePropsContext, InferGetServerSidePropsType } from "next";
-import { getServerSession } from "next-auth/next";
-import { getProviders, signIn } from "next-auth/react";
+import { ClientSafeProvider, getProviders, signIn } from "next-auth/react";
 import Head from "next/head";
 import Image from "next/image";
 import Link from "next/link";
 import { useEffect, useState } from "react";
 import Button from "../../components/Button";
+import LoadingOverlay from "../../components/LoadingOverlay";
 import { getProviderLogoSrc } from "../../components/ProviderLogo";
+import { useToast } from "../../components/Toast/use-toast";
 import VibinexDarkLogo from '../../public/vibinex-dark-logo.png';
 import VibinexLightLogo from '../../public/vibinex-light-logo.png';
 import type { RepoProvider } from "../../utils/providerAPI";
 import { getPreferredTheme, type Theme } from "../../utils/theme";
-import { authOptions } from "../api/auth/[...nextauth]";
 
-const SignInPage = ({ providers }: InferGetServerSidePropsType<typeof getServerSideProps>) => {
+const SignInPage = () => {
 	const vibinexLogo = (theme: Theme) => theme === 'dark' ? VibinexDarkLogo : VibinexLightLogo;
+	const { toast } = useToast();
+
+	const DEFAULT_CALLBACK_URL = "/u";
+	const callbackUrlReceived = typeof window !== "undefined" ? new URL(window.location.href).searchParams.get("callbackUrl") : null;
+	const callbackUrl = callbackUrlReceived ?? DEFAULT_CALLBACK_URL;
+
+	const [loading, setLoading] = useState(true);
 	const [theme, setTheme] = useState<Theme>('light');
+	const [providers, setProviders] = useState<ClientSafeProvider[]>([]);
 
 	useEffect(() => {
+		setLoading(true);
 		setTheme(getPreferredTheme());
-	}, [])
+
+		// get the list of providers
+		getProviders()
```

Comment:
so `getProviders` is a function in-built in next, right?

Reply:
Yes, in NextAuthJS. (not NextJS).